biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/configuration/BndConfiguration.java

```diff
@@ -47,6 +57,160 @@ public File loadProperties(Processor processor) throws Exception {
 		return loadProjectProperties(processor, project, project, configuration);
 	}
 
+	public void loadImplicitProperties(Processor processor, String outputTimestamp) throws Exception {
```

Comment:
Perhaps a better method name like `setPropertiesDefaults`. The method is not loading anything. It is setting default values for some bnd properties based upon the values of bnd properties and maven project properties.

It is unfortunate that you pass outputTimestamp here. It would be nicer if this value could be found through the plugin configuration since we may need to pass more such values in the future and end up with having to overload this method.

Reply:
> Perhaps a better method name like setPropertiesDefaults. The method is not loading anything. It is setting default values for some bnd properties based upon the values of bnd properties and maven project properties.

I was just inspired by the existing method name, that in also not necessarily "loads" something but just reads it from the project config and set the values... maybe `inheritPropertiesDefaults` would even be better.

> It is unfortunate that you pass outputTimestamp here. It would be nicer if this value could be found through the plugin configuration since we may need to pass more such values in the future and end up with having to overload this method.

Should be doable.

Reply:
I have now applied the suggestion to read the property from the configuration.